operator-common/src/main/java/io/strimzi/operator/common/controller/AbstractControllerLoop.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.controller;
+
+import io.micrometer.core.instrument.Timer;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.metrics.ControllerMetricsHolder;
+
+import java.util.TimerTask;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Abstract controller loop provides the shared functionality for reconciling resources in Strimzi controllers. It takes
+ * an event from a queue passed in controller and reconciles it.
+ */
+public abstract class AbstractControllerLoop implements Runnable {
```

Comment:
By implementing Runnable you expose a `run()` method to clients which they should never call (you expect them to instead follow the pattern of instantiate, `start()`, add to the queue..., `stop()`). So it would be better from an API design PoV to use a private inner class that implements Runnable to avoid ending up with a public `run()` that you never want anyone to call.